catalog/resource_system_schema.go

```diff
@@ -0,0 +1,116 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.org/x/exp/slices"
+)
+
+func ResourceSystemSchema() *schema.Resource {
+	create := func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+		o, n := d.GetChange("system_schema")
+		old, okOld := o.([]any)
+		new, okNew := n.([]any)
+		if !okNew || !okOld {
+			return fmt.Errorf("internal type casting error")
+		}
+		var oldSchemas, newSchemas []string
+		for _, schema := range old {
+			oldSchemas = append(oldSchemas, schema.(string))
+		}
+		for _, schema := range new {
+			newSchemas = append(newSchemas, schema.(string))
+		}
+		log.Printf("[DEBUG] Old system schemas: %v, new: %v", oldSchemas, newSchemas)
+		w, err := c.WorkspaceClient()
+		if err != nil {
+			return err
+		}
+		metastoreSummary, err := w.Metastores.Summary(ctx)
+		if err != nil {
+			return err
+		}
+		//enable new schemas that is not enabled
+		for _, schema := range newSchemas {
+			if !slices.Contains(oldSchemas, schema) {
+				err := w.SystemSchemas.Enable(ctx, catalog.EnableRequest{
+					MetastoreId: metastoreSummary.MetastoreId,
+					SchemaName:  catalog.EnableSchemaName(schema),
+				})
+				if err != nil {
+					return err
+				}
+			}
+		}
+		//disable old schemas that is not needed
+		for _, schema := range oldSchemas {
+			if !slices.Contains(newSchemas, schema) {
+				err := w.SystemSchemas.DisableByMetastoreIdAndSchemaName(ctx, metastoreSummary.MetastoreId, catalog.DisableSchemaName(schema))
+				if err != nil {
+					return err
+				}
+			}
+		}
+		d.SetId(metastoreSummary.MetastoreId)
+		return nil
+	}
+	return common.Resource{
+		Schema: map[string]*schema.Schema{
+			"system_schema": {
+				Type:     schema.TypeList,
```

Comment:
Wouldn't it be better to use `TypeSet` to avoid changes when order changes, handling duplicate elements, etc.?